src/stability_sdk/animation.py

```diff
@@ -420,16 +422,46 @@ def generate_depth_image(self, image: Image.Image) -> Image.Image:
         return results[0]
 
     def get_animation_prompts_weights(self, frame_idx: int) -> Tuple[List[str], List[float]]:
+        prev, next, tween = self.get_key_frame_tween(frame_idx)
+        if prev == next or not self.args.interpolate_prompts:
+            return [self.animation_prompts[prev]], [1.0]
+        else:
+            return [self.animation_prompts[prev], self.animation_prompts[next]], [1.0 - tween, tween]
+
+    def get_color_match_image(self, frame_idx: int) -> Image.Image:
+        if not self.args.color_match_animate:
+            return self.color_match_images.get(0)
+
+        prev, next, tween = self.get_key_frame_tween(frame_idx)
+
+        if prev not in self.color_match_images:
+            self.color_match_images[prev] = self._span_render_frame(prev, self.args.seed)
+        prev_match = self.color_match_images[prev]
+        if prev == next:
+            return prev_match
+
+        if next not in self.color_match_images:
+            self.color_match_images[next] = self._span_render_frame(next, self.args.seed)
+        next_match = self.color_match_images[next]
+
+        blended = prev_match.copy()
+        width, height, tile_size = blended.width, blended.height, 64
+        for y in range(0, height, tile_size):
+            for x in range(0, width, tile_size):
+                cut = next_match.crop((x, y, x + int(tile_size * tween), y + int(tile_size * tween)))
```

Comment:
Hm, why not just the weighted sum of `prev_match` and `next_match`? 

And maybe add an in-code comment also, because it doesn't seem obvious.

Reply:
That would introduce new colors in between the palettes instead of transitioning from colors of one to colors of the other. The blended image here has only RGB values that are present in either no new colors introduced this way.